packages/backend-common/src/database/SingleDatabase.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Knex from 'knex';
+import { Config } from '@backstage/config';
+import { createDatabaseClient, ensureDatabaseExists } from './connection';
+import {
+  DatabaseConfiguration,
+  PluginDatabaseFactory,
+  PluginDatabaseManager,
+} from './types';
+
+export class SingleDatabaseConfiguration implements DatabaseConfiguration {
+  private readonly config: Config;
+
+  constructor(config: Config) {
+    this.config = config;
+  }
+
+  getDatabaseConfig(_pluginId: string): Config {
+    return this.config;
+  }
+
+  getAdminConfig(): Config {
+    return this.config;
+  }
+}
+
+export class SingleDatabaseManager implements PluginDatabaseManager {
```

Comment:
What are you thoughts on naming this something like `SingleConnectionManager`?

Reply:
I'm OK with that, but if anyone else thinks we need a name please comment here. I'll change it to `SingleConnectionManager` first (which itself is a misnomer, since the connection pool isn't shared between plugins...)